Guide for running Kubeflow pipelines (#25)
dream-aim-deliver/planckster-docs: docs/guides/scrapers/Sentinel_setup.md

### For Linux:

1. **Update Your Package Index**:
   - Open a terminal and run:
     ```bash
     sudo apt-get update
     ```

2. **Install Required Packages**:
   - Run:
     ```bash
     sudo apt-get install apt-transport-https ca-certificates curl software-properties-common
     ```

3. **Add Docker’s Official GPG Key**:
   - Run:
     ```bash
     curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
     ```

4. **Add Docker Repository**:
   - Run:
     ```bash
     sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
     ```

5. **Install Docker Engine**:
   - Run:
     ```bash
     sudo apt-get update
     sudo apt-get install docker-ce
     ```

6. **Start Docker**:
   - Run:
     ```bash
     sudo systemctl start docker
     ```

7. **Verify Installation**:
   - Run:
     ```bash
     docker --version
     ```
   - You should see the Docker version information if it is installed correctly.

By following these steps, you will ensure that Docker is installed and running on your system.
**Ensure docker is installed and running succesfully before proceeding with the next steps**

> ### 2. Ensure you have a running instance of Kernel-Planckster , see [Guides](https://dream-aim-deliver.github.io/planckster-docs/docs/category/guides) -> [Kernel-planckster](https://dream-aim-deliver.github.io/planckster-docs/docs/category/kernel-planckster) for help.

Reviewer: ```suggestion
### 2. Ensure you have a running instance of Kernel-Planckster, see [Guides](https://dream-aim-deliver.github.io/planckster-docs/docs/category/guides) -> [Kernel-planckster](https://dream-aim-deliver.github.io/planckster-docs/docs/category/kernel-planckster) for help.
```

### 3. Obtain a client ID and client secret from [Sentinel Hub](https://www.sentinel-hub.com/).

**Optional: Create a Virtual environment to avoid any package version conflicts**
```bash
python3 -m venv .venv
source .venv/bin/activate
```
- `python3 -m venv`: This command uses the venv module in Python to create a virtual environment.
- `.venv`: This is the name of the directory where the virtual environment will be created. You can name it anything, but .venv is a common convention.

## Setup Instructions

### 1. Clone the Repository
```bash
git clone https://github.com/dream-aim-deliver/mpi-sda-sentinel.git
cd mpi-sda-sentinel
```
- `git clone https://github.com/dream-aim-deliver/mpi-sda-sentinel.git`:
    - `git clone`: This command is used to create a local copy of a remote repository.

>     - https://github.com/dream-aim-deliver/mpi-sda-sentinel.git: This is the URL of the remote Git repository you want to clone. It points to   the repository hosted on GitHub.
> 
> - When you run this command, Git downloads the entire repository (including its history and all files) to a directory on your local machine. The  directory will have the same name as the repository (mpi-sda-sentinel).

Reviewer: ```suggestion
    - https://github.com/dream-aim-deliver/mpi-sda-sentinel.git: This is the URL of the remote Git repository you want to clone. It points to the repository hosted on GitHub.

- When you run this command, Git downloads the entire repository (including its history and all files) to the directory on your local machine from which you ran the clone command. The new directory will have the same name as the repository (mpi-sda-sentinel).
```

- `cd mpi-sda-sentinel`:
    - `cd`: This command is used to change the current directory in your terminal or command prompt.
    - mpi-sda-sentinel: This should be the directory name you want to navigate into. Verify the correct directory name after cloning.

In summary, these commands clone a GitHub repository to your local machine and then change into the directory of the cloned repository.

### 2. Prepare Environment variables
Copy the `env.template` file to create a `.env` file:
```bash
cp .env.template .env
```
Fill in the environment variables in the `.env` file:
```bash
- sh_client_id = {ENTER THE CLIENT ID FROM Sentinel Hub}
- sh_client_secret = {ENTER CLIENT SECRET FROM Sentinel Hub}
- HOST = {THE HOSTNAME OF THE FASTAPI APP}
- PORT = {THE PORT OF THE FASTAPI APP}
```

### 3. Run the Docker Container
To build and run the Docker Container, execute the following script:
```bash
./run.sh
```
**troubleshoot common error:** The `run.sh` file has the following configurations
```bash
docker run --name mpi-satellite \
    --rm \
    -e "HOST=0.0.0.0" \
    -e "PORT=8000" \
    -e "sh_client_id=35c2e8f8-8796-450a-9770-f4e481949986" \

>     -e "sh_client_secret=Y*X?HpB9W4N&#lWKt!1CQL3cj-)r(6M{n}Nuj6QD"  \

Reviewer: Just checking, these are not real credentials, right? 😅

Author: yep, actually they were, it just slipped my mind, resolved it. :)

    -p "8000:8000" \
    mpi-satellite
```

> change the `PORT` to any other available port(example 8001) and the `-p` flag (to correspondingly "8001:8001") , if any port conflict occurs.

Reviewer: ```suggestion
change the `PORT` to any other available port (e.g. 8001) and the `-p` flag correspondingly ("8001:8001"), if any port conflict occurs.
```
